### ISSUE TITLE
proof2' doesn't pass type checking

### ISSUE THREAD

**Issue by grouzen** [liamoc/learn-you-an-agda#17]
Hi, I've been starting to read the book, and I can't get a code working:
```agda

proof₂′ : (A : Set) → A → A
proof₂′ _ x = x

proof₂ : (succ (succ zero)) even  → (succ (succ zero)) even
proof₂ x = proof₂′ ℕ x

```

The error that I've got:
> (succ (succ zero) even) !=< ℕ of type Set
> when checking that the expression x has type ℕ

Probably I'm missing something, or there is a typo in the book

I have Agda 2.4.2.2

Thanks,
Michael

**Comment by cjenkin2** [comment]
@grouzen 

The error message you are getting is more helpful than you realize. Allow me to explain:

The type of x in proof₂ is "(succ (succ zero)) even". When you call proof₂′ however, you are telling it that the type is ℕ. These two types are not the same, so it gives you a type error.

BTW: proof₂′ is the identity function, and generally you do not need to pass in the argument explicitly.

```agda
id : {A : Set} → A → A
id a = a
```

**Comment by grouzen** [comment]
@cjenkin2 Yeah! I noticed this too (types are different at all), that's why I created this issue.
